src/v0/analytics.ts

```diff
@@ -381,13 +386,48 @@ export class Analytics {
   getCustomersReport(query?: CustomersOptions | undefined): Promise<AnalyticsResponse> {
     return new Promise(async (resolve, reject) => {
       try {
-        let uri = `${this.options.base}${this.endpoint}/${this.options.user}/reports/customers`
+        const base = this.uriHelper.generateBaseUri('/reports/customers')
+        const uri = this.uriHelper.generateUriWithQuery(base, query)
 
-        const queryString = qs.stringify(query)
+        const response = await this.http.getClient().get(uri)
+        response.status !== 200 && reject(new errors.CustomerFetchFailed())
 
-        if (queryString) {
-          uri = `${uri}?${queryString}`
-        }
+        return resolve({
+          data: response.data.results,
+          metadata: { count: response.data.count }
+        } as AnalyticsResponse)
+      } catch (err) {
+        return reject(new errors.CustomerFetchFailed())
```

Comment:
Just wondering if the name of the error is correct? I was expecing something with "Analytics" in the name.